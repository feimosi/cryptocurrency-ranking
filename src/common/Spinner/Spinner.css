.Spinner {
  font-size: 10px;
  text-indent: -9999em;
  border-top: 2px solid rgba(255, 255, 255, 0.2);
  border-right: 2px solid rgba(255, 255, 255, 0.2);
  border-bottom: 2px solid rgba(255, 255, 255, 0.2);
  border-left: 2px solid #ffffff;
  transform: translateZ(0);
  animation: spinner 1.1s infinite linear;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  display: inline-block;
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
