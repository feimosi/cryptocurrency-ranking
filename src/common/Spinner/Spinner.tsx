import * as React from 'react';
import cn from 'classnames';

import PureBaseComponent from 'common/PureBaseComponent';
import './Spinner.css';

interface Props {
  className?: string;
  color?: string;
  size?: number;
}

export default class Spinner extends PureBaseComponent<Props> {
  render() {
    const {
      className,
      color = '#ffffff',
      size = 16,
    } = this.props;

    const classNames = cn(className, 'Spinner');
    const style = {
      width: `${size}px`,
      height: `${size}px`,
      borderTopWidth: `${size / 8}px`,
      borderRightWidth: `${size / 8}px`,
      borderBottomWidth: `${size / 8}px`,
      borderLeftWidth: `${size / 8}px`,
      borderLeftColor: color,
    };

    return (
      <span className={ classNames } style={ style } />
    );
  }
}
