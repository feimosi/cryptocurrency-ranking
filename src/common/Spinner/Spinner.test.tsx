import * as React from 'react';
import { shallow } from 'enzyme';

import Spinner from './Spinner';

describe('Spinner', () => {
  it('should shallow render correctly', () => {
    const wrapper = shallow(<Spinner />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should apply custom class name', () => {
    const className = 'Test__spinner';
    const wrapper = shallow(<Spinner className={ className } />);

    expect(wrapper.hasClass(className)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
