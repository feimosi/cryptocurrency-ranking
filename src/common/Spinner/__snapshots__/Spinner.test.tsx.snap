// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should apply custom class name 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    className="Test__spinner"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "Test__spinner Spinner",
      "style": Object {
        "borderBottomWidth": "2px",
        "borderLeftColor": "#ffffff",
        "borderLeftWidth": "2px",
        "borderRightWidth": "2px",
        "borderTopWidth": "2px",
        "height": "16px",
        "width": "16px",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Test__spinner Spinner",
        "style": Object {
          "borderBottomWidth": "2px",
          "borderLeftColor": "#ffffff",
          "borderLeftWidth": "2px",
          "borderRightWidth": "2px",
          "borderTopWidth": "2px",
          "height": "16px",
          "width": "16px",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Spinner should shallow render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "Spinner",
      "style": Object {
        "borderBottomWidth": "2px",
        "borderLeftColor": "#ffffff",
        "borderLeftWidth": "2px",
        "borderRightWidth": "2px",
        "borderTopWidth": "2px",
        "height": "16px",
        "width": "16px",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Spinner",
        "style": Object {
          "borderBottomWidth": "2px",
          "borderLeftColor": "#ffffff",
          "borderLeftWidth": "2px",
          "borderRightWidth": "2px",
          "borderTopWidth": "2px",
          "height": "16px",
          "width": "16px",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
